DOCS-460: Adding tutorial content for Plan Motion with an Arm and a Gripper (#873)
viamrobotics/docs: docs/tutorials/services/plan-motion-with-arm-gripper.md

}
```

{{% /tab %}}
{{< /tabs >}}

For the gripper pose, you can change the reference frame information to consider other objects or user-generated frames that exist in the Frame System.
Specifying other reference frames is an easy way to move with respect to those frames.
For example, you can specify a pose that is 100 millimeters above the table obstacle featured earlier in this tutorial.
You do not need to calculate that exact pose with respect to the **arm** or **world**.
You must only provide the object name (instead of the `gripperName` you saw in the code samples above) when making the `NewPoseInFrame` to pass into the `Move` function.
This has implications for how motion is calculated, what magnitude and direction is considered, and what the mot to consider when planning.

## Next Steps and References

<!-- TODO: Content below struck out for the moment, saved to point at the next tutorial "Plan Motion with an Arm and with a Gripper" -->
<!--
If you would like to continue onto specific complex motion constraints, go to the next tutorial in this series: "[Put Constraints on a Motion Plan](/tutorials/motion/plan-motion-with-arm-gripper/)".
-->

{{< snippet "social.md" >}}

> <!-- TODO : Add more content here -->

Author: This could be struck out, or we could come up with some more clever links to reference frame content or something else thats meaningful. Feel free to bring up ideas for what would fit here nicely.

Author: Actually, this comment is defunct after having reviewed your other feedback

All you need is a robot with a component that can move, such as a robotic arm.

## Access the Motion Service

Accessing the Motion Service is very similar to accessing any other component or service within the Viam ecosystem.

{{< tabs >}}
{{% tab name="Python" %}}
You must import an additional Python library to access the Motion Service.
Add `from viam.services.motion import MotionServiceClient` to your import list and then add the sample code below to your own client script.

```python {class="line-numbers linkable-line-numbers"}
# Access the Motion Service
motion_service = MotionServiceClient.from_robot(robot, "builtin")
```

{{% /tab %}}
{{% tab name="Go" %}}
You must import an additional Go package to access the Motion Service.
Add `"go.viam.com/rdk/services/motion"` to your import list and then add the sample code below to your own client script.

```go {class="line-numbers linkable-line-numbers"}
motionService, err := motion.FromRobot(robot, "builtin")
if err != nil {
  logger.Fatal(err)
}
```

{{% /tab %}}
{{< /tabs >}}

Once the Motion Service can be accessed, some familiar features become available.

> The Motion service has a method that can get the *pose* of a component relative to a **reference frame**.

Reviewer: ```suggestion
The Motion service has a method that can get the *pose* of a component relative to a *reference frame*.
```

obstacles := make([]spatialmath.Geometry, 0)

tableOrigin := spatialmath.NewPose(
  r3.Vector{X: 0.0, Y: 0.0, Z: -10.0},
  &spatialmath.OrientationVectorDegrees{OX: 0.0, OY: 0.0, OZ: 1.0, Theta: 0.0},
)
tableDims := r3.Vector{X: 2000.0, Y: 2000.0, Z: 20.0}
tableObj, _ := spatialmath.NewBox(tableOrigin, tableDims, "table")
obstacles = append(obstacles, tableObj)

// Create a WorldState that has the GeometriesInFrame included
obstaclesInFrame := referenceframe.NewGeometriesInFrame(referenceframe.World, obstacles)
worldState := &referenceframe.WorldState{
  Obstacles: []*referenceframe.GeometriesInFrame{obstaclesInFrame},
}
```

{{% /tab %}}
{{< /tabs >}}

This example adds a "table" with the assumption that you mounted your robot arm to an elevated surface.
The 2000 millimeter by 2000 millimeter dimensions ensure that a sufficiently large box is constructed, regardless of the real physical footprint of your mounting surface.
Feel free to change these dimensions, including thickness (the Z coordinate in the above code samples), to match your environment more closely.
Additional obstacles can also be *appended* as desired.

{{< alert title="Tip" color="note" >}}

> Within the App, the **Frame System** tab in the **Config** section of your robot gives you the ability to experiment with various geometry representations with better visual feedback.

Reviewer: ```suggestion
Within the app, the **Frame System** tab in the **CONFIG** section of your robot gives you the ability to experiment with various geometry representations with better visual feedback.
```

Because of ongoing experimental API changes, we must pass in a `slam` service resource name when using the Motion service to `Move`.
Add `"go.viam.com/rdk/services/slam"` to your import list to provide access to the [SLAM Service](/services/slam).
This tutorial will not cover any other SLAM content.

```go {class="line-numbers linkable-line-numbers"}
// Generate a sample "start" pose to demonstrate motion
testStartPose := spatialmath.NewPose(
  r3.Vector{X: 510.0, Y: 0.0, Z: 526.0},
  &spatialmath.OrientationVectorDegrees{OX: 0.7071, OY: 0.0, OZ: -0.7071, Theta: 0.0},
)
testStartPoseInFrame := referenceframe.NewPoseInFrame(referenceframe.World, testStartPose)

_, err = motionService.Move(context.Background(), myArmResource, testStartPoseInFrame, worldState, nil, slam.Named(""), nil)
if err != nil {
  logger.Fatal(err)
}
```

{{% /tab %}}
{{< /tabs >}}

<!-- TODO : In the future, we should add some specific information on the importance of the frame chosen as the point of reference for PoseInFrame variables -->
<!-- ## Managing Pose References -->

## Command other components to move with the Motion Service

This section will require you to add a new component to your robot.

> One device that is very commonly attached to the end of a robot arm is a **gripper**.

Reviewer: ```suggestion
One device that is very commonly attached to the end of a robot arm is a *gripper*.
```

> Most robot arms pick up and manipulate objects in the world through the use of a gripper, so learning how to directly command a gripper to move is highly valuable.

Reviewer: ```suggestion
Most robot arms pick up and manipulate objects in the world through the use of a gripper, so learning how to directly move a gripper is very useful.
```
[opt]

Though various Motion Service commands cause the gripper to move, ultimately the arm is doing all of the work in these situations.

> This is possible because the Motion Service considers other parts of the robot (through the [Frame System](/services/frame-system/) when calculating how to achieve the desired motion.

Reviewer: ```suggestion
This is possible because the Motion Service considers other components of the robot (through the [Frame System](/services/frame-system/) when calculating how to achieve the desired motion.
```
since "part" is a Viam reserved word. Could also say "pieces" if you want I guess

Author: Components works
